executor/analyze.go

```diff
@@ -111,7 +113,20 @@ type analyzeTask struct {
 	colExec  *AnalyzeColumnsExec
 }
 
-func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultCh chan<- statistics.AnalyzeResult) {
+func (e *AnalyzeExec) analyzeWorker(taskCh chan *analyzeTask, resultCh chan<- statistics.AnalyzeResult) {
+	defer func() {
+		if r := recover(); r != nil {
+			buf := make([]byte, 4096)
+			stackSize := runtime.Stack(buf, false)
+			buf = buf[:stackSize]
+			log.Errorf("analyzeWorker panic stack is:\n%s", buf)
+			metrics.PanicCounter.WithLabelValues(metrics.LabelAnalyze).Inc()
+			// Close the receive channel. When the caller sends to a closed channel, it will
+			// also panic, and the panic should be catched. If we don't do this, the caller
+			// will hang forever.
+			close(taskCh)
```

Comment:
Should we also close the `resultCh`? Should we also recover the panic in `Next` function?